AutomatedTesting/Gem/PythonTests/Physics/tests/EntityComponentTests/PhysX_Collider_Component_CRUD.py

```diff
@@ -0,0 +1,167 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+def PhysX_Collider_Component_CRUD():
+    """
+    Summary:
+    Creating a PhysX Collider and setting values to all properties that are supported via the Editor Context Bus.
+
+    Expected Behavior:
+    Validates that Component CRUD can be performed without crashing.
+
+    PhysX Collider Components:
+     - PhysX Collider
+
+    Test Steps:
+     1) Add an Entity to manipulate
+     2) Add a PhysX Collider Component to Entity
+     3) Set Box Shape and Child Properties
+     4) Set Capsule Shape and Child Properties
+     5) Set Cylinder Shape and Child Properties
+     6) Set Sphere Shape and Child Properties
+     7) Set General Properties
+     8) Set PhyicsAsset Shape and Child Properties
+     9) Delete Component
+
+    :return: None
+    """
+
+    # Helper file Imports
+    import os
+
+    from editor_python_test_tools.editor_component.physx_collider import PhysxCollider
+    from editor_python_test_tools.editor_entity_utils import EditorEntity
+    from editor_python_test_tools.utils import Tracer, TestHelper
+
+    from consts.general import Strings
+
+    # 0) Pre-conditions
+    physx_mesh = os.path.join("objects", "_primitives", "_box_1x1.pxmesh")
+    physx_material = os.path.join("physx", "glass.physxmaterial")
+
+    TestHelper.init_idle()
+    TestHelper.open_level("", "Base")
+
+    with Tracer() as section_tracer:
+    # 1 ) Add an Entity to manipulate
+        phsyx_collider_entity = EditorEntity.create_editor_entity("TestEntity")
+
+    # 2) Add a PhysX Collider Component to Entity
+        physx_collider = PhysxCollider(phsyx_collider_entity)
+
+    # 3) Set Box Shape and Child Properties
+        physx_collider.set_box_shape()
```

Comment:
all these setters and toggles tests are missing to check if the value are actually set or toggled.

I would expect something like this in here:

````
physx_collider.set_box_shape()
# assert if physx_collider.get_shape() is not box shape
````

````
physx_collider.set_box_dimensions(256.0, 256.0, 256.0)
# assert if physx_collider.get_box_dimensions() is not (256.0, 256.0, 256.0)
````

Reply:
There's a validation set assert that occurs within `EntityCollider.set_component_property_value(...)` which is the function that is called with formed data.

    ```
def set_component_property_value(self, component_property_path: str, value: object):
        """
        Used to set component property value
        :param component_property_path: Path of property in the component to act on
        :param value: new value for the variable being changed in the component
        """
        outcome = editor.EditorComponentAPIBus(
            bus.Broadcast, "SetComponentProperty", self.id, component_property_path, value
        )
        assert (
            outcome.IsSuccess()
        ), f"Failure: Could not set value to '{self.get_component_name()}' : '{component_property_path}'"
        PrefabWaiter.wait_for_propagation()
        self.get_property_tree(True)
```

While this doesn't specifically validate that the bus correctly set the data, it does validate that the bus returned that it was successful at setting the data.

I've thought about doing data set validation additionally, even wrote a few functions that way, but made the call that it would potentially be redundant.

@moraaar if you're still of the opinion of doing an explicit data-is-accurately-set validation, I can add that in.

Reply:
Yeah, I would add the explicit data check too. As you say at the moment it only checks if the bus call `SetComponentProperty` returned success, but that's not enough.

On example from the top of my head is that maybe the edit context for a property has extra parameters, like limits that should apply to values, and those must be checked if they are applied correctly.

````
physx_collider.set_whatever_property(-1) // let's assume there are limits min=0 and max=1
assert physx_collider.get_whatever_property() == 0 // This getter needs to return the expected value within limits to test if the limits work
````

Reply:
I think you should create a validation helper file rather than packing validation into the set_component_property_value.
basically a test helper that is given a value or list of values then it does the set, get, and verify outside of the editor_entity_utils which is just about abstracting the ebus interactions with properties.
with your component class system you can even create something that does a validate_property_values(func property_setter, list_values)
that itself can call the property_setter method in your class with each value in the list and validate that it gets the value back. you could special case by type validate_float_property(func, list) which would then include the Math_IsClose to handle float precision in the background.

Reply:
Refactored all of this logic.